src/interpreter/mod.rs

```diff
@@ -0,0 +1,492 @@
+use std::{collections::HashMap, hash::Hash};
+
+use num_bigint::BigInt;
+
+use crate::{
+    compiler::{semantic::SymTable, Message},
+    field::Field,
+    interpreter::{expr::eval_expr, frame::StackFrame},
+    parser::ast::{
+        expression::Expression, statement::Statement, tl::TLDecl, DebugSymRef, Identifiable,
+        Identifier,
+    },
+    wit_gen::{StepInstance, SymbolSignalMapping},
+};
+
+use self::value::Value;
+
+mod expr;
+mod frame;
+mod value;
+
+struct Interpreter<'a, F: Field + Hash> {
+    mapping: &'a SymbolSignalMapping,
+    cur_frame: StackFrame<'a, F>,
+
+    witness: Vec<StepInstance<F>>,
+}
+
+impl<'a, F: Field + Hash> Interpreter<'a, F> {
+    pub fn new(symbols: &'a SymTable, mapping: &'a SymbolSignalMapping) -> Self {
+        Self {
+            mapping,
+            cur_frame: StackFrame::new(symbols),
+            witness: Vec::default(),
+        }
+    }
+
+    pub fn run(
```

Comment:
Let's add docs to public functions. Is the input a mapping of input signal names to values? (that's my guess)